packages/components/src/components/Checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,263 @@
+    paddingTop: spacing.vadsSpaceSm,
+    paddingBottom: spacing.vadsSpaceSm,
+    paddingLeft: spacing.vadsSpaceSm,
+    paddingRight: spacing.vadsSpaceMd,
+    borderColor: checked
+      ? theme.vadsColorFormsBorderActive
+      : theme.vadsColorFormsBorderSubtle,
+    backgroundColor: checked
+      ? theme.vadsColorFormsSurfaceActive
+      : theme.vadsColorSurfaceDefault,
+  }
+
+  /**
+   * Icon
+   */
+
+  const iconViewStyle: ViewStyle = {
+    alignItems: 'center',
+    justifyContent: 'center',
+    paddingTop: 3,
+    paddingBottom: 3,
+  }
+
+  const _icon = (
+    <View style={iconViewStyle}>
+      <Icon
+        name={
+          indeterminate
+            ? 'IndeterminateCheckBox'
+            : checked
+              ? 'CheckBox'
+              : 'CheckBoxOutlineBlank'
+        }
+        fill={
+          checked
+            ? theme.vadsColorFormsForegroundActive
+            : theme.vadsColorFormsBorderDefault
+        }
+      />
+    </View>
+  )
+
+  /**
+   * Label
+   */
+
+  const labelProps = {
+    'aria-label': labelA11y || label,
+    style: {
+      ...baseTextStyle,
+      fontFamily: error ? 'SourceSansPro-Bold' : 'SourceSansPro-Regular',
+      fontSize: 20,
+      lineHeight: 30,
+    },
+  }
+
+  const requiredStyle = {
+    color: theme.vadsColorForegroundError,
+  }
+
+  const _label = (
+    <Text {...labelProps}>
+      {label}
+      {required && <Text style={requiredStyle}>{` (*${t('required')})`}</Text>}
+    </Text>
+  )
+
+  /**
+   * Header
+   */
+
+  const headerProps = {
+    'aria-label': headerA11y || header,
+    style: { ...baseTextStyle, fontSize: 20, lineHeight: 30 },
+  }
+
+  const _header = header && (
+    <>
+      <Text {...headerProps}>{header}</Text>
+      <Spacer size="xs" />
+    </>
+  )
+
+  /**
+   * Hint
+   */
+
+  const hintProps = {
+    'aria-label': hintA11y || hint,
+    style: {
+      ...baseTextStyle,
+      color: theme.vadsColorForegroundSubtle,
+      fontSize: 16,
+      lineHeight: 22,
+    },
+  }
+
+  const _hint = hint && (
+    <>
+      <Text {...hintProps}>{hint}</Text>
+      <Spacer size="xs" />
+    </>
+  )
+
+  /**
+   * Error
+   */
+
+  const errorProps = {
+    'aria-label': `${t('error')}: ${errorA11y || error}`,
+    style: {
+      fontFamily: 'SourceSansPro-Bold',
+      color: theme.vadsColorForegroundError,
+      fontSize: 16,
+      lineHeight: 22,
+    },
+  }
+
+  const _error = error && (
+    <>
+      <Text {...errorProps}>{error}</Text>
+      <Spacer size="xs" />
+    </>
+  )
+
+  /**
+   * Description
+   */
+
+  const descriptionProps = {
+    'aria-label': descriptionA11y || description,
+    style: { ...baseTextStyle, fontSize: 16, lineHeight: 22 },
+  }
+
+  const _description = description && (
+    <>
+      <Spacer size="xs" />
+      <Text {...descriptionProps}>{description}</Text>
+    </>
+  )
+
+  return (
+    <ComponentWrapper>
+      <View style={containerStyle}>
+        {_header}
+        {_hint}
```

Comment:
Not sure I want to see it changed, but just throwing out that it's more difficult to follow the Spacer use having it nested within the various pieces. But it also may be obnoxious to have all these separated by `_header && _hint ? <Spacer > : null`.

I definitely like the general principle of splitting out these named parts though as it helps to see the high level pertinent structure.